config/config_test.go

```diff
@@ -142,3 +144,62 @@ func TestMergeGlobalConfig(t *testing.T) {
 	// The log level should now be debug.
 	assert.Equal(t, "debug", config.Global.Loggers[Default].Level)
 }
+
+// initializeConfig initializes the configuration with the given context.
+// It returns a pointer to the Config struct. If configuration initialization fails,
+// the test will fail with an error message.
+func initializeConfig(ctx context.Context, t *testing.T) *Config {
+	t.Helper()
+	config := NewConfig(ctx, Config{
+		GlobalConfigFile: parentDir + GlobalConfigFilename,
+		PluginConfigFile: parentDir + PluginsConfigFilename,
+	})
+	err := config.InitConfig(ctx)
+	require.Nil(t, err)
+	return config
+}
+
+// serverLoadBalancerStrategyOverwrite sets the environment variable for server nested configuration
+// and verifies that the configuration is correctly loaded with the expected value.
+func ServerLoadBalancerStrategyOverwrite(t *testing.T) {
+	t.Helper()
+	ctx := context.Background()
+	// Convert to uppercase
+	upperDefaultGroup := strings.ToUpper(Default)
+
+	// Format environment variable name
+	envVarName := fmt.Sprintf("GATEWAYD_SERVERS_%s_LOADBALANCER_STRATEGY", upperDefaultGroup)
+
+	// Set the environment variable
+	t.Setenv(envVarName, "test")
+	config := initializeConfig(ctx, t)
+	assert.Equal(t, "test", config.Global.Servers[Default].LoadBalancer.Strategy)
+}
+
+// pluginDefaultPolicyOverwrite sets the environment variable for plugin configuration
+// and verifies that the configuration is correctly loaded with the expected value.
+func pluginDefaultPolicyOverwrite(t *testing.T) {
+	t.Helper()
+	ctx := context.Background()
+
+	// Set the environment variable
+	t.Setenv("GATEWAYD_DEFAULTPOLICY", "test")
```

Comment:
I would have expected


```suggestion
	t.Setenv("GATEWAYD_DEFAULT_POLICY", "test")
```

Is there a problem or something to code differently?

Reply:
This is a test case for the plugin. In the plugin configuration, there isn't a `configuration group`, so `DefaultPolicy` is a parameter directly within `gatewayd_plugins.yaml`.
```
CompatibilityPolicy: "strict"
DefaultPolicy: "passthrough"
```